tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -92,6 +92,8 @@ static void counter_tear_down_instance(const char *dev_name)
 
 	err = counter_stop(dev);
 	zassert_equal(0, err, "Counter failed to stop\n");
+
+	alarm_cnt = 0;
```

Comment:
This should be done in `test_all_channels_instance` where it is actually needed. Doing it here is obscure.